extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpHostConfigSource.java

```diff
@@ -45,19 +45,11 @@ public String getValue(String propertyName) {
             // In dev-mode we want to only listen on localhost so others on the network cannot connect to the application.
             // However, in WSL this would result in the application not being accessible,
             // so in that case, we launch it on all interfaces.
-            return (LaunchMode.current().isDevOrTest() && !isWSL()) ? "localhost" : ALL_INTERFACES;
+            return (LaunchMode.current().isDevOrTest() && !LaunchMode.isWSL()) ? "localhost" : ALL_INTERFACES;
         }
         return null;
     }
 
-    /**
-     * @return {@code true} if the application is running in a WSL (Windows Subsystem for Linux) environment
-     */
-    private boolean isWSL() {
-        var sysEnv = System.getenv();
-        return sysEnv.containsKey("IS_WSL") || sysEnv.containsKey("WSL_DISTRO_NAME");
```

Comment:
@dmlloyd @radcortez I wonder if it's something we should include in SmallRye Common (OS maybe?)?

Reply:
I created https://github.com/smallrye/smallrye-common/issues/311 .